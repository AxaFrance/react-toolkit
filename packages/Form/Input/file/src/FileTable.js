import React from 'react';
import PropTypes from 'prop-types';
import { ClassManager } from '@axa-fr/react-toolkit-core';
import LineFile from './FileLine';

const propTypes = {
  values: PropTypes.array,
  errors: PropTypes.array,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  className: PropTypes.string,
  classModifier: PropTypes.string,
};
const defaultClassName = 'custom-table-file af-file-table';
const defaultProps = {
  values: null,
  errors: null,
  disabled: false,
  className: defaultClassName,
  classModifier: null,
};

const Errors = errors => {
  if (!errors || errors.length <= 0) {
    return null;
  }

  const errorList = errors.map(err =>
    err ? (
      <li key={err}>
        {err.file.name} ({err.file.size})
      </li>
    ) : (
      ''
    )
  );
  return (
    <div className="af-form__file-errors">
      <span>Le chargement de certains fichiers a échoué : </span>
      <ul className="af-form__file-errors-list">{errorList}</ul>
    </div>
  );
};

const FileTable = ({
  errors,
  values,
  className,
  classModifier,
  disabled,
  onClick,
}) => {
  if (!values || values.length <= 0) {
    return null;
  }
  const componentClassName = ClassManager.getComponentClassName(
    className,
    classModifier,
    defaultClassName
  );
  return (
    <div className={componentClassName}>
      {Errors(errors)}
      <ul className="af-form__file-list">
        {values.map(({ file, id }) => (
          <LineFile
            disabled={disabled}
            file={file}
            onClick={onClick}
            id={id}
            key={id}
          />
        ))}
      </ul>
    </div>
  );
};

FileTable.propTypes = propTypes;
FileTable.defaultProps = defaultProps;

export default FileTable;
