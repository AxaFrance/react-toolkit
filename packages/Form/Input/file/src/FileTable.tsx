import React, { ComponentProps, ComponentPropsWithoutRef } from 'react';
import { ClassManager } from '@axa-fr/react-toolkit-core';
import LineFile from './FileLine';

const defaultClassName = 'custom-table-file af-file-table';

const Errors = ({ errors }: Pick<Props, 'errors'>) => {
  return (
    <div className="af-form__file-errors">
      <span>Le chargement de certains fichiers a échoué : </span>
      <ul className="af-form__file-errors-list">
        {errors.map((err) => (
          <li key={err.file.name}>
            {err.file.name} ({err.file.size})
          </li>
        ))}
      </ul>
    </div>
  );
};

type FileLineAction = Pick<ComponentProps<typeof LineFile>, 'onClick'>;
type File = Pick<ComponentPropsWithoutRef<typeof LineFile>, 'file'>;
type Props = FileLineAction & {
  errors?: File[];
  values?: (File & { id: string })[];
  className?: string;
  classModifier?: string;
  disabled?: boolean;
};
const FileTable = ({
  errors,
  values,
  className = defaultClassName,
  classModifier,
  disabled,
  onClick,
}: Props) => {
  const componentClassName = ClassManager.getComponentClassName(
    className,
    classModifier,
    defaultClassName
  );
  return (
    <div className={componentClassName}>
      {errors && errors.length > 0 && <Errors errors={errors} />}
      {values && values.length > 0 && (
        <ul className="af-form__file-list">
          {values.map(({ file, id }) => (
            <LineFile
              disabled={disabled}
              file={file}
              onClick={onClick}
              id={id}
              key={id}
            />
          ))}
        </ul>
      )}
    </div>
  );
};

export default FileTable;
