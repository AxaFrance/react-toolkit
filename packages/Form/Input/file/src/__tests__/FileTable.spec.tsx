import React from 'react';
import { render } from '@testing-library/react';
import FileTable from '../FileTable';

describe('<File.FileTable>', () => {
  it('renders File.FileTable correctly', () => {
    const { asFragment } = render(
      <FileTable
        errors={[{ file: { name: 'youhou.txt', preview: '', size: 2 } }]}
        values={[
          { file: { name: 'superfile.pdf', preview: '', size: 2 }, id: 'id' },
        ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders File.FileTable correctly when no have values', () => {
    const { asFragment, queryByRole } = render(
      <FileTable
        errors={[{ file: { name: 'youhou.txt', size: 2, preview: '' } }]}
        values={[]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(queryByRole('list')).toHaveClass('af-form__file-errors-list');
    expect(queryByRole('list')).toBeInTheDocument();
  });

  it('renders Errors correctly with empty errors', () => {
    const { asFragment, queryByRole } = render(
      <FileTable
        values={[
          { file: { name: 'youhou.txt', size: 2, preview: '' }, id: 'id' },
        ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(queryByRole('list')).not.toHaveClass('af-form__file-errors-list');
  });
});
