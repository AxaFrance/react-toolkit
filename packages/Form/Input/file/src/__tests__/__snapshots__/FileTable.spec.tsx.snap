// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<File.FileTable> renders Errors correctly with empty errors 1`] = `
<DocumentFragment>
  <div
    class="custom-table-file af-file-table"
  >
    <ul
      class="af-form__file-list"
    >
      <li
        class="af-form__file-line"
      >
        <i
          class="glyphicon glyphicon-file"
        />
        <span>
          youhou.txt
        </span>
        <span>
          2
        </span>
        <button
          class="af-link af-link--delete-file"
          type="button"
        >
          <span
            class="af-link__text"
          >
            Supprimer
          </span>
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<File.FileTable> renders File.FileTable correctly 1`] = `
<DocumentFragment>
  <div
    class="custom-table-file af-file-table"
  >
    <div
      class="af-form__file-errors"
    >
      <span>
        Le chargement de certains fichiers a échoué : 
      </span>
      <ul
        class="af-form__file-errors-list"
      >
        <li>
          youhou.txt (2)
        </li>
      </ul>
    </div>
    <ul
      class="af-form__file-list"
    >
      <li
        class="af-form__file-line"
      >
        <i
          class="glyphicon glyphicon-file"
        />
        <span>
          superfile.pdf
        </span>
        <span>
          2
        </span>
        <button
          class="af-link af-link--delete-file"
          type="button"
        >
          <span
            class="af-link__text"
          >
            Supprimer
          </span>
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<File.FileTable> renders File.FileTable correctly when no have values 1`] = `
<DocumentFragment>
  <div
    class="custom-table-file af-file-table"
  >
    <div
      class="af-form__file-errors"
    >
      <span>
        Le chargement de certains fichiers a échoué : 
      </span>
      <ul
        class="af-form__file-errors-list"
      >
        <li>
          youhou.txt (2)
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
